### ISSUE TITLE
[v1.1] Model weights and gradients not synchronized in Model Adapter Doamin Adapter

### ISSUE THREAD

**Issue by zhouyu5** [intel/e2eAIOK#197]
- When running ddp training with domain adapter, see no grads sync after each backward process, which is different from what is said in this [tutorial](https://pytorch.org/tutorials/intermediate/ddp_tutorial.html)

    > When the backward() returns, param.grad already contains the synchronized gradient tensor.

- It is likely to be a bug, which can validate with following scripts:
```python
for name, params in self.network.named_parameters():
    if params.grad is None: continue
    print(f'rank: {dist.get_rank()}, name: {name}, grads: {torch.sum(params.grad)}, value: {torch.sum(params)}')
sys.exit(0)
```
